test/TEST-16-DMSQUASH/test-init.sh

```diff
@@ -9,6 +9,10 @@ exec > /dev/console 2>&1
 
 echo "dracut-root-block-success" | dd oflag=direct,dsync of=/dev/disk/by-id/ata-disk_marker
 
+if grep -qF ' rd.live.overlay=LABEL=persist ' /proc/cmdline; then
+    echo "dracut-autooverlay-success" > /overlay-marker
```

Comment:
Is there a benefit of using / partition as marker and not /dev/disk/by-id/ata-disk_marker partition like other tests ?

Reply:
This ensures that the overlay is actually working. The test now creates the root partition as only half the size of the image file, leaving the second half blank. When it boots, the autooverlay module creates a partition in the free space and uses it for the overlay. Writing to a file in the root filesystem lets the test verify that the autooverlay module successfully created and formatted the partition and that the dmsquash-live module used it when setting up the rootfs overlay. The test script verifies that the string `dracut-autooverlay-success` occurs in the second half of the image file, I.E. within the second partition.

Reply:
Thanks, would love to see this explanation in the code as a comment :-) 
Also I was hoping that reviewing how the test is expected to work helps you figure out why the test is failing on the CI ...

Reply:
I finally had a chance to add comments explaining the reasoning and look into the test failure today.

Here is TEST-16-DMSQUASH, now passing in CI in my fork using all container variants except Gentoo:
* Arch: https://github.com/iammattcoleman/dracut/actions/runs/3253893225/jobs/5341593145
* Debian: https://github.com/iammattcoleman/dracut/actions/runs/3253902964/jobs/5341613095
* Fedora: https://github.com/iammattcoleman/dracut/actions/runs/3253910178/jobs/5341627573
* OpenSuse: https://github.com/iammattcoleman/dracut/actions/runs/3253919573/jobs/5341644938

Gentoo is failing with a node-related error during the "Checkout Repository" stage that I don't understand:
https://github.com/iammattcoleman/dracut/actions/runs/3253852146/jobs/5341510695

I found this ticket that describes a similar failure, but it's specifically about a self-hosted runner and unfortunately doesn't offer any details about how they fixed it:
https://github.com/actions/checkout/issues/749

Reply:
After investigating a bit more, I see that you've (@LaszloGombos) come across this issue with the Gentoo container, as well: https://github.com/dracutdevs/dracut/pull/1812#issuecomment-1263954331

It's caused by an architecture mismatch as described in this comment: https://github.com/actions/checkout/issues/334#issuecomment-1241306390

Here's an issue in actions/runner that describes how the runner uses hardcoded paths for node:
https://github.com/actions/runner/issues/2115

As suggested in the I reverted to actions/checkout@v1 and it revealed that the Gentoo container is missing dependencies still:
https://github.com/iammattcoleman/dracut/actions/runs/3254015105/jobs/5341833491

I'm fixing that now.

Reply:
I fixed the issues with Gentoo.

Here is TEST-16-DMSQUASH, now passing in CI in my fork using all container variants:
* Arch: https://github.com/iammattcoleman/dracut/actions/runs/3254211434/jobs/5342208863
* Debian: https://github.com/iammattcoleman/dracut/actions/runs/3254212012/jobs/5342210044
* Fedora: https://github.com/iammattcoleman/dracut/actions/runs/3254212447/jobs/5342211082
* Gentoo: https://github.com/iammattcoleman/dracut/actions/runs/3254221018/jobs/5342224579
* OpenSuse: https://github.com/iammattcoleman/dracut/actions/runs/3254221363/jobs/5342225163